test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Xml.Tests.ps1

```diff
@@ -0,0 +1,138 @@
+		$xmlContent = @"
+<Root>
+   <Node Attribute='blah' />
+</Root>
+"@
+
+		Set-Content -Path $fileName -Value $xmlContent
+
+		$fileNameWithDots = $fileName.FullName.Replace("\", "\.\")
+
+		$driveLetter = [string]($fileName.FullName)[0]
+		$fileNameAsNetworkPath = "\\localhost\$driveLetter`$" + $fileName.FullName.SubString(2)
+
+		$testCases = @(
+			@{testName = 'Literalpath with relative paths'; testParameter = @{LiteralPath = $fileName.Name; XPath = 'Root'}},
+			@{testName = 'Literalpath with absolute paths'; testParameter = @{LiteralPath = $fileName.FullName; XPath = 'Root'}},
+			@{testName = 'Literalpath with path with dots'; testParameter = @{LiteralPath = $fileNameWithDots; XPath = 'Root'}},
+			@{testName = 'Path with relative paths'; testParameter = @{Path = $fileName.Name; XPath = 'Root'}},
+			@{testName = 'Path with absolute paths'; testParameter = @{Path = $fileName.FullName; XPath = 'Root'}},
+			@{testName = 'Path with path with dots'; testParameter = @{Path = $fileNameWithDots; XPath = 'Root'}}
+		)
+
+		if ( ! $IsCoreCLR ) {
+			$testcases += @{testName = 'Literalpath with network paths'; testParameter = @{LiteralPath = $fileNameAsNetworkPath; XPath = 'Root'}}
+			$testcases += @{testName = 'Path with network paths'; testParameter = @{LiteralPath = $fileNameAsNetworkPath; XPath = 'Root'}}
+		}
+
+		Push-Location -Path $fileName.Directory
+	}
+
+	AfterAll {
+		Pop-Location
+	}
+
+	It "Can work with input files using <testName>" -TestCases $testCases {
+		param($testParameter)
+
+		$node = Select-Xml @testParameter
+		$node | Should -HaveCount 1
+		$node.Path | Should -Be $fileName.FullName
+	}
+
+	It "Can work with input streams" {
+		[xml]$xml = "<a xmlns='bar'><b xmlns:b='foo'>hello</b><c>world</c></a>"
+		$node = Select-Xml -Xml $xml -XPath "//c:b" -Namespace @{c='bar'}
+		$node.Path | Should -BeExactly "InputStream"
+		$node.Pattern | Should -BeExactly "//c:b"
+		$node.ToString() | Should -BeExactly "hello"
+	}
+
+	It "Can throw on non filesystem paths using <parameter>" -TestCases @(
+		@{parameter="LiteralPath"; expectedError='ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'},
+		@{parameter="Path";        expectedError='ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'}
+	) {
+		param($parameter, $expectedError)
+		try
+		{
+			$env:xmltestfile = $xmlContent
+			$file = 'env:xmltestfile'
+			$params = @{$parameter=$file}
+			$err = $null
+			Select-Xml @params "Root" -ErrorVariable err -ErrorAction SilentlyContinue
+			$err.FullyQualifiedErrorId | Should -Be $expectedError
+		}
+		finally
+		{
+			Remove-Item -Path 'env:xmltestfile'
+		}
+	}
+
+	It "Can throw for invalid XML file" {
+		$testfile = "TestDrive:\test.xml"
+		Set-Content -Path $testfile -Value "<a><b>"
+		$err = $null
+		Select-Xml -Path $testfile -XPath foo -ErrorVariable err -ErrorAction SilentlyContinue
+		$err.FullyQualifiedErrorId | Should -Be 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'
```

Comment:
The recommended pattern is to use `Should -Throw -ErrorId`. Have a look at: https://github.com/PowerShell/PowerShell/blob/a62eb62701091222a0adade43806e4fba97bae7b/docs/testing-guidelines/WritingPesterTests.md#writing-pester-tests

Reply:
Thank you for the great suggestion @adityapatwardhan